docs/proposals/disable-fencing.txt

```diff
@@ -0,0 +1,31 @@
+Disable Machine Fencing 
+=========================
+
+Summary
+----------------
+
+The Machine Healthcheck Controller (MHC) looks for failed nodes and deletes them after giving them grace time to recover (5 min by default). This feature provides a mechanism to prevent that from happening on a per-node basis.
+
+
+Motivation 
+----------------
+
+The administrator should be able to exclude nodes from being fenced.This will allow to track the machine manually without applying the machine fencing mechanism according to the system administrator considerations.
```

Comment:
I am confused by the "fencing" term. In my understanding:
Fencing is the process of isolating a node to protect a cluster and its resources. Without fencing, a faulty node can cause data corruption in a cluster.

Your proposal looks like disabling machine healthchecking. is machine healthcking and fencing same thing?

Reply:
I find the terminology confusing too.

We talk about health checking (noticing a failure) and remediation (currently just deletion).

I believe the point of both health checking and the remediation is (a) to allow safely moving workloads from unhealthy nodes and (b) attempting to bring the node back into a healthy state. And so we say this is fencing.

But if we're going to use that term at all, we should use it consistently.

e.g. README.md says:

> Machine healthcheck controller
> Reconciles desired state for MachineHealthChecks by ensuring that machines targeted by machineHealthCheck objects are healthy or remediated otherwise

Would we be happy to change that to:

> Machine healthcheck (fencing) controller
> Provides node level fencing by checking that machines targeted by MachineHealthCheck objects are healthy (as indicated by NodeConditions) and attempting remediation of unhealthy machines to bring them back to a healthy state